Test: ringbuffer: Improving the description for ringbuffer test cases.

Zhaoningx opened the pull request:
The existing testcase's doxygen describes are the general
implementation idea of a function.On this basis, adding
more descriptive statements to describe which conditions need
to be preset when running the testcase, which test techniques are applied,
and describe the testcase Design steps in detail. Make it more readable.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

LeiW000 commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
+ * - N/A
+ *
+ * Test Procedure:
+ * -# defined an array with some data items that ready for being
```
Please capitalize the first letter.

LeiW000 commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
+ * Test Procedure:
+ * -# defined an array with some data items that ready for being
+ * put.
+ * -# Do a while loop to put data items.
```
It better to change it like below.

"Put data with "while loop"".

LeiW000 commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
+ * -# defined an array with some data items that ready for being
+ * put.
+ * -# Do a while loop to put data items.
+ * -# Check if an error will be met when the ringbuffer is full.
```
It better to change "an error will be met" with "an error will be seen".

LeiW000 commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
+ * -# Get data items from the ringbuffer.
+ * -# Check if the data put are equal to the data got.
+ * -# Going on getting data from the ringbuffer.
+ * -# Check if an error will be met when the ringbuffer is empty.
```
Same as line 70.

LeiW000 commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
+ * -# Check if an error will be met when the ringbuffer is empty.
  *
+ * Expected Test Result:
+ * - Data items that pushed will be equal to data items that got and
```
It better to chnage it like this

Data items pushed shall be equal to what are gotten.

LeiW000 commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
  *
+ * Expected Test Result:
+ * - Data items that pushed will be equal to data items that got and
+ * an error will be met when you try to put a item into a full ringbuffer or
```
It better to change it like:

An error shall be shown up when an item is put into a full ringbutter

LeiW000 commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
+ * get some items from an empty ringbuffer.
+ *
+ * Pass/Fail Criteria:
+ * - Success if test results of step 3,5,7 of condition have all meet.
```
It better to change it like:

"Success if test result of step 3.5,7 is passed. Otherwise, failure."

Then, line 83 can be removed.

Zhaoningx replied:
Hi @LeiW000 , Thanks for your advice, I have changed them.

Commit:
Tests: ringbuffer: Improve the description cases

The existing testcase's doxygen describes are the general
implementation idea of a function.On this basis, adding
more descriptive statements to describe which conditions
need to be preset when running the testcase, which test
techniques are applied, and describe the testcase Design
steps in detail. Make it more readable.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

enjiamai commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
+ * - Define and initialize a ringbuffer by using macro
+ * RING_BUF_ITEM_DECLARE_POW2
+ *
+ * Input Specifications:
```
Please descript the input for these API briefly, it should have.

enjiamai commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -186,6 +220,46 @@ void test_ringbuffer_declare_size(void)
+ * - Define a pointer of ring buffer type.
+ *
+ * Input Specifications:
+ * - N/A
```
Please descript the input for these API briefly,

enjiamai commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -234,6 +353,51 @@ void test_ringbuffer_pow2_put_get_thread_isr(void)
+ * - Define a pointer of ring buffer type.
+ *
+ * Input Specifications:
+ * - N/A
```
Please descript the input for these API briefly,

enjiamai commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -246,16 +410,44 @@ void test_ringbuffer_size_put_get_thread_isr(void)
+ * - Define and initialize a ringbuffer by RING_BUF_DECLARE
+ *
+ * Input Specifications:
+ * - N/A
```
Please descript the input for these API briefly,

enjiamai commented on tests/lib/ringbuffer/src/main.c:
```diff
@@ -476,15 +666,46 @@ static uint32_t ringbuf_stored[RINGBUFFER_SIZE];
+ * - Define an array that changes as the system changes.
+ *
+ * Input Specifications:
+ * - N/A
```
Please descript the input for these API briefly,

Zhaoningx replied:
Hi @enjiamai ,after discussing, this part needn't to modify.

enjiamai replied:
Ok, no problem.

enjiamai reviewed (changes requested):
Please refine the input parameter for target testing function if any, thanks!

LeiW000 commented:
Is there any other comment? If not, I think it can be merged. It is just doxygen comment change.
